src/main/java/gyro/google/storage/BucketLifecycleRuleAction.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.storage;
+
+import com.google.api.services.storage.model.Bucket.Lifecycle.Rule.Action;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+
+/**
+ * The action to take.
+ */
+public class BucketLifecycleRuleAction extends Diffable implements Copyable<Action> {
+
+    private String storageClass;
+    private String type;
+
+    /**
+     * Target storage class. Required when ``action`` is set to ``SetStorageClass``.
+     */
+    @Updatable
+    public String getStorageClass() {
+        return storageClass;
+    }
+
+    public void setStorageClass(String storageClass) {
+        this.storageClass = storageClass;
+    }
+
+    /**
+     * Type of the action to take on condition. Valid types are ``Delete`` or ``SetStorageClass``.
+     */
+    @Updatable
```

Comment:
This shouldn't be updatable as this field is part of the primary key and any change triggers a create / delete flow

Reply:
Removed annotation